### PR TITLE
[IMP] purchase_request: make mail sending optional

### PR DESCRIPTION
**Pull request by i-vyshnevska**

### CODE REVIEW

**Comment by ajaniszewska-dev** [comment]
can we update readme about this functionality?

**Review by mmequignon — APPROVED** [review]

**Review by simahawk — CHANGES_REQUESTED** [review]
pls document this new config in `readme/CONFIGURE.rst`

**Review by HviorForgeFlow — APPROVED** [review]

**Comment by HviorForgeFlow** [comment]
@i-vyshnevska could you squash fixup commit? Thanks

File: purchase_request/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -7,3 +7,6 @@ To configure the product follow this steps:
 With this configuration, whenever a procurement order is created and the supply
 rule selected is 'Buy' the application will create a Purchase Request instead
 of a Purchase Order.
+
+When delivery to purchase request is confirmed notification to partner is generated.
+Can be disabled in configuration settings.
```
Comment (simahawk) [inline purchase_request/readme/CONFIGURE.rst]:
```suggestion
When a delivery for purchase request is confirmed, a notification to followers is generated.
You can control this with the flag "Notify followers on request allocation" in stock general settings
```

File: purchase_request/views/res_config_settings.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                    <div class="o_setting_left_pane">
+                        <field name="notify_request_allocations" />
+                    </div>
+                    <div class="o_setting_right_pane">
+                        <label
+                            for="notify_request_allocations"
+                            string="Request mail options"
+                        />
+                        <div class="text-muted">
+                            Notify folowers on request allocation
```
Comment (simahawk) [inline purchase_request/views/res_config_settings.xml]:
```suggestion
                            Notify followers on request allocation
```

**Review by simahawk — CHANGES_REQUESTED** [review]
Please squash when done.

**Comment by CasVissers-360ERP** [comment]
@i-vyshnevska @HviorForgeFlow Is a forward port is this planned to v15? The current functionality is very strict? Lots of companies do not want this kind of automatic communication.

EDIT: Nevermind, I had an issue in the followers set-up.

**Pull request merged**